crates/sui-framework/packages/sui-framework/tests/coin_tests.move

```diff
@@ -648,4 +649,97 @@ module sui::coin_tests {
         transfer::public_freeze_object(deny_cap);
         scenario.end();
     }
+
+    #[test]
+    fun test_is_regulated_currency() {
+        let ctx = &mut tx_context::dummy();
+        deny_list::create_for_test(ctx);
+
+        let witness = COIN_TESTS {};
+        let (treasury, deny_cap, metadata) = coin::create_regulated_currency(
+            witness,
+            6,
+            b"COIN_TESTS",
+            b"coin_name",
+            b"description",
+            option::some(url::new_unsafe_from_bytes(b"icon_url")),
+            ctx,
+        );
+        assert!(coin::is_regulated_currency(&metadata).is_some());
+        assert!(coin::is_regulated_currency(&metadata).extract());
+        assert!(!coin::allows_global_pause(&metadata).extract());
+        test_utils::destroy(treasury);
+        test_utils::destroy(deny_cap);
+        test_utils::destroy(metadata);
+    }
+
+
+    #[test]
+    fun test_is_regulated_currency_v2_global_pause() {
+        let ctx = &mut tx_context::dummy();
+        deny_list::create_for_test(ctx);
+
+        let witness = COIN_TESTS {};
+        let (treasury, deny_cap, metadata) = coin::create_regulated_currency_v2(
+            witness,
+            6,
+            b"COIN_TESTS",
+            b"coin_name",
+            b"description",
+            option::some(url::new_unsafe_from_bytes(b"icon_url")),
+            true,
+            ctx,
+        );
+        assert!(coin::is_regulated_currency(&metadata).is_some());
+        assert!(coin::is_regulated_currency(&metadata).extract());
+        assert!(coin::allows_global_pause(&metadata).extract());
+        test_utils::destroy(treasury);
+        test_utils::destroy(deny_cap);
+        test_utils::destroy(metadata);
+    }
+
+    #[test]
+    fun test_is_regulated_currency_v2_no_global_pause() {
+        let ctx = &mut tx_context::dummy();
+        deny_list::create_for_test(ctx);
+
+        let witness = COIN_TESTS {};
+        let (treasury, deny_cap, metadata) = coin::create_regulated_currency_v2(
+            witness,
+            6,
+            b"COIN_TESTS",
+            b"coin_name",
+            b"description",
+            option::some(url::new_unsafe_from_bytes(b"icon_url")),
+            false,
+            ctx,
+        );
+        assert!(coin::is_regulated_currency(&metadata).is_some());
```

Comment:
Blind (untested) comment but receiver syntax should work, no? They are both in the coin module (can be applied to all tests / calls).
```suggestion
        assert!(metadata.is_regulated_currency().is_some());
```